sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -907,6 +902,12 @@ case class MergeIntoTable(
       case _ => false
     }
   }
+
+  override def left: LogicalPlan = targetTable
+  override def right: LogicalPlan = sourceTable
+  override protected def withNewChildrenInternal(
+      newLeft: LogicalPlan, newRight: LogicalPlan): MergeIntoTable =
+    copy(targetTable = newLeft, sourceTable = newRight)
```

Comment:
Hi, @szehon-ho . May I ask why you propose to move the above 5 lines as an optimization? This doesn't look like an optimization. If this is a simple relocation, I'd ask you revert this change from this PR because this PR's title is `Small optimization for ...`.

Reply:
Hi @dongjoon-hyun thanks for looking at it, yes its a minor cleanup, i noticed the other lazy vals are above the left, right, etc.  I can revert it tomorrow if its an issue

Reply:
Got it. No problem. Let me revise the PR title and description a little and merge this, @szehon-ho. Thank you.